### PR TITLE
Story 32: Admin Invoices Index Page

### LINKED ISSUE #60: 32. Admin Invoices Index Page
As an admin,
When I visit the admin Invoices index (/admin/invoices)
Then I see a list of all Invoice ids in the system
Each id links to the admin invoice show page

### PR DESCRIPTION
**Pull request by JorjaF**
__x__ Wrote Tests ____ Implemented ____ Reviewed

Neccesary checkmarks:

    [x] All Tests are Passing

    [x] The code will run locally

Type of change

    [x] New feature
    [] Bug Fix

Implements/Fixes: completes story 32 and adds admin invoice index page

    description closes #60 

Check the correct boxes

    [x] This broke nothing
    [] This broke some stuff
    [] This broke everything

Testing Changes

    [x] No Tests have been changed
    [] Some Tests have been changed
    [] All of the Tests have been changed(Please describe what in the world happened)

Checklist:

    [x] My code has no unused/commented out code
    [x] I have reviewed my code
    [] I have commented my code, particularly in hard-to-understand areas
    [x] I have fully tested my code

(REQUIRED)Please Include a link to a gif of your feelings about this branch

Link: https://giphy.com/clips/buzzfeed-buzzfeed-celeb-kids-do-movie-sound-effects-NQb5W7wXqiI04aL8nt

### CODE REVIEW

**Comment by ethanrossblack** [comment]
32. Admin Invoices Index Page

As an admin,
When I visit the admin Invoices index (/admin/invoices)
Then I see a list of all Invoice ids in the system
Each id links to the admin invoice show page

File: app/views/admin/invoices/index.html.erb [hunk 1]
```diff
@@ -0,0 +1,11 @@
+<div class="invoices-header">
+  <h2> Invoices </h2>
+</div>
```
Comment (ethanrossblack) [inline app/views/admin/invoices/index.html.erb]: Maybe incorporate the partial header that Daniel made?

File: spec/features/admin/admin_invoice_spec.rb [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+# US 32
+  describe "As an admin when I visit the admin invoices page" do
+
+    it "I can see a list of all invoice ids in the system" do
+
+      visit admin_invoices_path
+
+      expect(Invoice.all.count).to eq(6) 
+      
+      Invoice.all.each do |invoice|
```
Comment (ethanrossblack) [inline spec/features/admin/admin_invoice_spec.rb:25]: Very smart! I should definitely start using iterations more in my testing to DRY things up.

File: spec/features/admin/admin_invoice_spec.rb [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+  end
+
+# US 32
+  describe "As an admin when I visit the admin invoices page" do
+
+    it "I can see a list of all invoice ids in the system" do
+
+      visit admin_invoices_path
+
+      expect(Invoice.all.count).to eq(6)
```
Comment (ethanrossblack) [inline spec/features/admin/admin_invoice_spec.rb]: I don't think you need this line. We're testing that the invoices show up on the page, not the invoice count.

File: spec/features/admin/admin_merchant_spec.rb [hunk 4]
```diff
@@ -19,16 +19,17 @@
...
-    click_link("Bob's Burgers")
-    expect(current_path).to eq(admin_merchant_path(@merchant_1))
+      visit admin_merchants_path
+    
+      click_link("Bob's Burgers")
+      expect(current_path).to eq(admin_merchant_path(@merchant_1))
+    end
+
+    it "Has two sections, one for enabled merchants and one for disabled merchants" do
+
```
Comment (ethanrossblack) [inline spec/features/admin/admin_merchant_spec.rb]: How did this get in the User Story 32 test?

File: app/views/admin/invoices/index.html.erb [hunk 5]
```diff
@@ -0,0 +1,11 @@
...
+<div class="invoices">
+  <% @invoices.each do |invoice| %>
+    <div class="invoice">
+      <%= link_to invoice.id, admin_invoice_path(invoice) %>
```
Comment (ethanrossblack) [inline app/views/admin/invoices/index.html.erb:6]: In the wireframe, it shows each invoice number has `Invoice #` before each invoice id/link. I think you should add that.

Here's how I wrote the HTML for the Merchant Invoices Index page, which is very similar visually.

```
<div id="merchant_invoices">
  <h1>My Invoices</h1>
  <% @merchant.distinct_invoices.each do |invoice| %>
    <p>Invoice #<%= link_to "#{invoice.id}", merchant_invoice_path(@merchant, invoice), id: "invoice#{invoice.id}_link" %></p>
  <% end %>
</div>
```

image

**Review by ethanrossblack — CHANGES_REQUESTED** [review]

File: app/views/admin/invoices/index.html.erb (hunk 1 above)
Comment (JorjaF) [reply app/views/admin/invoices/index.html.erb]: I tried to use that partial but it does not seem to work. when i do it i get an error that says 
```
Missing partial admin/invoices/_admin_header, application/_admin_header with {:locale=>[:en], :formats=>[:html], :variants=>[], :handlers=>[:raw, :erb, :html, :builder, :ruby, :jbuilder]}.
```

File: spec/features/admin/admin_invoice_spec.rb (hunk 2 above)
Comment (JorjaF) [reply spec/features/admin/admin_invoice_spec.rb:25]: Well thank you!

File: app/views/admin/invoices/index.html.erb (hunk 5 above)
Comment (JorjaF) [reply app/views/admin/invoices/index.html.erb:6]: fixed

File: app/views/admin/invoices/index.html.erb (hunk 1 above)
Comment (Daniel-Gallagher92) [reply app/views/admin/invoices/index.html.erb]: As of now the _admin_header and the _new_edit partial work :)

File: spec/features/admin/admin_invoice_spec.rb (hunk 2 above)
Comment (Daniel-Gallagher92) [reply spec/features/admin/admin_invoice_spec.rb:25]: Nice!

File: spec/features/admin/merchants/index_spec.rb
Comment (Daniel-Gallagher92) [inline spec/features/admin/merchants/index_spec.rb:1]: Solid use of an ID vs. class here

**Review by Daniel-Gallagher92 — APPROVED** [review]
It made me laugh, but are we going to keep the #sanity check?

File: app/views/admin/invoices/index.html.erb (hunk 1 above)
Comment (JorjaF) [reply app/views/admin/invoices/index.html.erb]: I added this partial now

**Comment by JorjaF** [comment]
> It made me laugh, but are we going to keep the #sanity check?

> i deleted it

**Pull request closed without merging**